pvlib/pvsystem.py

```diff
@@ -2537,6 +2553,73 @@ def pvwatts_losses(soiling=2, shading=3, snow=0, mismatch=2, wiring=2,
     return losses
 
 
+def irradiance_loss_pvsyst(effective_irradiance, irradiance_shading_loss,
+                           irradiance_snow_loss, irradiance_soiling_loss):
```

Comment:
The function would return the same thing if the user permuted the loss arguments. We could consider `*losses` for a more general and extensible approach.

Reply:
I like this idea.  I believe calling out the losses explicitly helps guide new users, but I think an in-depth tutorial can serve that purpose better while allowing for more flexibility here.

If we make this update, I don't think the function name "irradiance_loss_pvsyst" is apt anymore.  The point of the function is that it allows a convenient way to compound any irradiance losses that should be applied to the total POA (not specific to beam, diffuse, or ground POA).  Maybe that means it should be called "compound_effective_irradiance_losses?"  Maybe "combine_effective_irradiance_losses?"

A future implementation more inline with PVSyst should be a "pvsyst_effective_irradiance" function.

Reply:
+1 to `combine_loss_factors` or similar for a general purpose function, since the PVWatts model does the same calculation on a different set of inputs.

I think it's helpful to also have a function named `irradiance_loss_pvsyst` or similar, with arguments that are easily recognized by someone used to using the PVsyst loss model. If we keep a _pvsyst function, it should return effective_irradiance with losses applied, instead of the loss factor.

`effective_irradiance_pvsyst` implies that the function also calculates and applies reflection and spectral mismatch modifiers, which is a larger scope that this PR.